### ISSUE TITLE
Cleanup usages of the membership table InstanceName field. Maybe rename to SiloName

### ISSUE THREAD

**Issue by shayhatsor** [dotnet/orleans#1738]
This continues the discussing started in #1725. A partial fix for the issue was implemented in #1728.
We still need to add the `InstanceName` field back to the relational implementation of the membership table. This raises some questions:
1) Do we rename `InstanceName` to `SiloName` ?
2) `InstanceName` is being used in some flows, we should investigate whether existing code considers `InstanceName` as being unique or non-empty or any other predicate.

**Comment by gabikliot** [comment]
I will look at 1 and 2 next week.

**Comment by sergeybykov** [comment]
👍

**Comment by gabikliot** [comment]
I renamed InstanceName to SiloName in #1740.

As for uniqueness:
1) **The membership protocol does not rely on uniqueness of SiloNames**. There is one place in membership where we do use SiloNames: detect silo migration function. We look at the silo name and if we find that silo with the same name moved to a different host and or a new IP, we use that as a heuristic to say: silo has migrated. The only thing we do in this case it log a message. No actual logic relies on that "migration detection".
2) There is another component called` IDeploymentConfiguration`. It just returns silo names. The interface by itself does not guarantee silo name uniqueness and all implementation arr (currently) not required to guarantee that. BUT, different implementations of it can guarantee that. And indeed Azure implementation of that interface (OrleansAzureUtils/Hosting/ServiceRuntimeWrapper.cs) guarantees uniqueness. 
3) Later on we have another Azure-specific component (Azure based  Deployment queue Balancer, used in Event Hub Stream Provider) which does heavily rely on the  uniqueness of SiloNames.

To summarize: as of now, there is no general requirement for all implementations of system store or deployment to provide unique silo name. 
Different implementations  of system store or deployment **can** guarantee  uniqueness, if they wish so. In such cases other components that take direct dependency on that specific implementation can rely on the guarantees uniqueness.
